esmlab/core.py

```diff
@@ -87,6 +87,7 @@ def compute_time(self):
         ds[self.time_coord_name] = groupby_coord.data
 
         if self.time_bound is not None:
+            ds[self.tb_name] = self.time_bound
```

Comment:
I started to investigate why `test_resample_mon_mean` was failing for you, and am a little confused about what this line is doing (or is supposed to do). It seems like `self.time_bound` is an array of `nan`s here, which is good in https://github.com/NCAR/esmlab/issues/161#issuecomment-548975319 because at least `nan` is a `numpy` type and not a `cftime` type but it's probably not the desired behavior and hence why the test is failing. Here's some additional output from my notebook (showing the initial value of `ds[self.tb_name]` and then what it was set to in this line):

```
ds[self.tb_name]: <xarray.DataArray 'time_bound' (time: 1980, d2: 2)>
dask.array<open_dataset-aa1aa6d7d49084e6d24552872c45da2ctime_bound, shape=(1980, 2), dtype=object, chunksize=(48, 2), chunktype=numpy.ndarray>
Coordinates:
  * time     (time) object 1850-01-16 13:00:00 ... 2014-12-16 12:00:00
Dimensions without coordinates: d2
Attributes:
    long_name:  boundaries for time-averaging interval
self.time_bound: <xarray.DataArray 'time_bound' (time: 1980, d2: 2)>
array([[nan, nan],
       [nan, nan],
       [nan, nan],
       ...,
       [nan, nan],
       [nan, nan],
       [nan, nan]])
Coordinates:
  * time     (time) object 1850-01-16 13:00:00 ... 2014-12-16 12:00:00
Dimensions without coordinates: d2
Attributes:
    long_name:  boundaries for time-averaging interval
```

And then the same from `test_core.py`:

```
E   ValueError: invalid second provided in cftime.DatetimeNoLeap(-5883517, 2, 27, 0, 0, -2147483648, -2147483648, 5, 58)

cftime/_cftime.pyx:1637: ValueError
------------------------------------------------------------- Captured stdout call -------------------------------------------------------------
ds[self.tb_name]: <xarray.DataArray 'time_bound' (time: 24, d2: 2)>
array([[  0.,  31.],
       [ 31.,  59.],
       [ 59.,  90.],
       [ 90., 120.],
       [120., 151.],
       [151., 181.],
       [181., 212.],
       [212., 243.],
       [243., 273.],
       [273., 304.],
       [304., 334.],
       [334., 365.],
       [365., 396.],
       [396., 424.],
       [424., 455.],
       [455., 485.],
       [485., 516.],
       [516., 546.],
       [546., 577.],
       [577., 608.],
       [608., 638.],
       [638., 669.],
       [669., 699.],
       [699., 730.]])
Coordinates:
  * time     (time) object 0001-01-16 12:00:00 ... 0002-12-16 12:00:00
  * d2       (d2) int64 0 1
self.time_bound: <xarray.DataArray 'time_bound' (time: 24, d2: 2)>
array([[nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan],
       [nan, nan]])
Coordinates:
  * time     (time) object 0001-01-16 12:00:00 ... 0002-12-16 12:00:00
  * d2       (d2) int64 0 1
```

I can keep digging if you want me to, but I'd guess that you're setting `self.time_bound` correctly for annual resampling and not for monthly (based on the tests that pass)

Reply:
Kind of ironic given how much time you've spend dropping `.data` from the lefthand side in this package, but

```
-            ds[self.tb_name] = self.time_bound
+            ds[self.tb_name].data = self.time_bound.data
```

Passes the CI tests and also avoids the `unsupported operand type(s)` that @klindsay28 and I were seeing.

My notebook is now hanging on a `esmlab.weighted_sum()` call, though I'm hoping that's just a coincidence and not an indication something else going wrong. I'll investigate more on Monday.

Reply:
> My notebook is now hanging on a `esmlab.weighted_sum()` call, though I'm hoping that's just a coincidence

That must've been something glade-related, my notebook works with the recommended addition of `.data`

Reply:
Thank you for looking into this, @mnlevy1981!